tomcat/src/main/java/com/linecorp/armeria/server/http/tomcat/TomcatService.java

```diff
@@ -100,6 +100,17 @@
         TOMCAT_URL_ENCODER.addSafeCharacter('/');
     }
 
+    static final TomcatHandler TOMCAT_HANDLER;
+
+    static {
+        // Load a tomcat handler from the providers.
+        ServiceLoader<TomcatHandlerProvider> loader =
+                ServiceLoader.load(TomcatHandlerProvider.class, TomcatHandlerProvider.class.getClassLoader());
+
+        checkState(loader.iterator().hasNext(), "Cannot find a usable TomcatHandlerProvider");
+        TOMCAT_HANDLER = checkNotNull(loader.iterator().next().entry(), "TomcatHandler");
```

Comment:
Wouldn't it be enough just to use reflection? This will expose `TomcatHandlerProvider` and potentially other classes to users. It would be just enough to determine the Tomcat version and to instantiate the classes for Tomcat 7.5/8.0 when the version is <= 8.0.